twtxt/helper.py

```diff
@@ -85,3 +87,28 @@ def sort_and_truncate_tweets(tweets, direction, limit):
             return sorted(tweets)
     else:
         return []
+
+
+@contextlib.contextmanager
+def atomic_write(path, mode):
+    """
+        Simple contextmanager to safely write to a
+        file.
+
+        :param str filename: the path to write to
+        :param str mode: the mode to open the file
+    """
+    # generate random filepath in same location as given filename
+    tmpfilepath = os.path.join(os.path.dirname(path), ".{}".format(uuid.uuid4()))
+    try:
+        if "a" in mode:
+            os.rename(path, tmpfilepath)
+
+        with open(tmpfilepath, mode) as atomicfile:
+            yield atomicfile
+        os.rename(tmpfilepath, path)
+    finally:
+        try:
+            os.unlink(tmpfilepath)
```

Comment:
Oh yeah, I see. The first os.rename would need to be *rolled back* .. somehow ..